src/collection.rs

```diff
@@ -35,8 +43,26 @@ impl Collection {
     ///
     /// * `path` - The path to the collection.
     pub fn from_path(path: PathBuf) -> Result<Self, ParseError> {
-        let file = BufReader::new(File::open(path)?);
-        let collection = Collection::try_from(&Element::parse(file)?)?;
+        Ok(Self::from_path_(path)?)
+    }
+
+    /// Create a new `Collection` from an XML file; skipping invalid components.
+    ///
+    /// For each invalid component, an error is collected into the returned
+    /// value.
+    ///
+    /// # Arguments
+    ///
+    /// * `path` - The path to the collection.
+    pub fn from_path_with_partial(
```

Comment:
Can we get a similar method for the gzipped variant please?

Reply:
Skipping this since I think they don't work anyways